tfhe/src/integer/server_key/radix_parallel/div_mod.rs

```diff
@@ -862,4 +874,146 @@ impl ServerKey {
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use tfhe::integer::gen_keys_radix;
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2;
+    ///
+    /// // Generate the client key and the server key:
+    /// let num_blocks = 4;
+    /// let (cks, sks) = gen_keys_radix(PARAM_MESSAGE_2_CARRY_2, num_blocks);
+    ///
+    /// let msg = 97u8;
+    ///
+    /// let ct1 = cks.encrypt(msg);
+    /// let ct2 = cks.encrypt(0u8);
+    ///
+    /// // Compute homomorphically an addition:
+    /// let (ct_q, ct_r, div_by_0) = sks.checked_div_rem_parallelized(&ct1, &ct2);
+    ///
+    /// // Decrypt:
+    /// let div_by_0 = cks.decrypt_bool(&div_by_0);
+    /// assert_eq!(div_by_0, true);
+    ///
+    /// let q: u8 = cks.decrypt(&ct_q);
+    /// assert_eq!(u8::MAX, q);
+    ///
+    /// let r: u8 = cks.decrypt(&ct_r);
+    /// assert_eq!(msg, r);
+    /// ```
+    pub fn checked_div_rem_parallelized<T>(
+        &self,
+        numerator: &T,
+        divisor: &T,
+    ) -> (T, T, BooleanBlock)
+    where
+        T: IntegerRadixCiphertext,
+    {
+        let ((q, r), div_by_0) = rayon::join(
+            || self.div_rem_parallelized(numerator, divisor),
+            || self.are_all_blocks_zero(divisor.blocks()),
+        );
+
+        (q, r, BooleanBlock::new_unchecked(div_by_0))
+    }
+
+    /// Computes homomorphically the quotient of the division between two ciphertexts
+    ///
+    /// Returns an additional flag indicating if the divisor was 0
+    ///
+    /// # Note
+    ///
+    /// If you need both the quotient and remainder use [Self::div_rem_parallelized].
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use tfhe::integer::gen_keys_radix;
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2;
+    ///
+    /// // Generate the client key and the server key:
+    /// let num_blocks = 4;
+    /// let (cks, sks) = gen_keys_radix(PARAM_MESSAGE_2_CARRY_2, num_blocks);
+    ///
+    /// let msg = 97u8;
+    ///
+    /// let ct1 = cks.encrypt(msg);
+    /// let ct2 = cks.encrypt(0u8);
+    ///
+    /// // Compute homomorphically an addition:
+    /// let (ct_res, div_by_0) = sks.checked_div_parallelized(&ct1, &ct2);
+    ///
+    /// // Decrypt:
+    /// let div_by_0 = cks.decrypt_bool(&div_by_0);
+    /// assert_eq!(div_by_0, true);
+    ///
+    /// let dec_result: u8 = cks.decrypt(&ct_res);
+    /// assert_eq!(u8::MAX, dec_result);
+    /// ```
+    pub fn checked_div_parallelized<T>(&self, numerator: &T, divisor: &T) -> (T, BooleanBlock)
+    where
+        T: IntegerRadixCiphertext,
+    {
+        let (q, div_by_0) = rayon::join(
+            || self.div_parallelized(numerator, divisor),
+            || self.are_all_blocks_zero(divisor.blocks()),
+        );
+
+        (q, BooleanBlock::new_unchecked(div_by_0))
+    }
+
+    /// Computes homomorphically the remainder (rest) of the division between two ciphertexts
+    ///
```

Comment:
should the divisor == 0 case be documented here as well ?

Reply:
I did not put the thing in the checked_ ops as, to me, if you are interested in that flag, it means you are going to use it, so you don't really care about the result in case divisor is 0

Reply:
fair enough